subsys/net/lib/sockets/Kconfig

```diff
@@ -48,7 +48,7 @@ config NET_SOCKETS_DNS_TIMEOUT
 
 config NET_SOCKETS_SOCKOPT_TLS
 	bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
-	select TLS_CREDENTIALS if NET_NATIVE
+	select TLS_CREDENTIALS
```

Comment:
Ok, I might have added this extra condition too hastily, based on the fact that nRF91 does not use it. It should be no harm for us to have it enabled, the linker will deal with it, but perhaps we could make this a less restrictive `imply TLS_CREDENTIALS`, so that it's still possible to disable it at Kconfig level?

Reply:
Updated to using 'imply'